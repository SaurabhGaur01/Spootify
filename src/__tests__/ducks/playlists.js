import {
    setPlaylistsData,
    clearPlaylistsData,
    playlistsDataReducer,
} from '../../ducks/playlists';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('playlists Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setPlaylistsData should match the snapshot' , () => {
            expect(setPlaylistsData(mockObject)).toMatchSnapshot();
        });
        it('clearPlaylistsData should match the snapshot' , () => {
            expect(clearPlaylistsData()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(playlistsDataReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(playlistsDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(playlistsDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should set the value when called with setPlaylistsData' , () => {
            expect(playlistsDataReducer('', setPlaylistsData(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearPlaylistsData' , () => {
            expect(playlistsDataReducer(mockObject, clearPlaylistsData())).toEqual([]);
        });
    });
});