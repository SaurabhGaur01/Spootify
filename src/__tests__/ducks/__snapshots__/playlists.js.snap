// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playlists Duck Tests Action Generator clearPlaylistsData should match the snapshot 1`] = `
Object {
  "type": "modules/spootify-app/CLEAR_PLAYLISTS_DATA",
}
`;

exports[`playlists Duck Tests Action Generator setPlaylistsData should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/spootify-app/SET_PLAYLISTS_DATA",
}
`;
